docs/content/scripting.md

```diff
@@ -119,4 +119,103 @@ from the `page` object.
 
 That's how you can use Rhai to add custom formatters to the site.
 
+## Building on the example
+
+When a site has a lot of blog posts, it would make sense to implement pagination for the content. This will introduce the idea of working with objects and nesting function calls. 
+
+Firstly, we will need need a script to parse the URL to obtain the page number to generate the pagination list.
+```rust
+let full_url = params[0];
+let query_params = #{page: 0};
+let query_string;
+let query_param;
+
+if full_url.contains("?") {
+    // Get the query string
+    query_string = full_url.sub_string(full_url.index_of("?") + 1);
+    do {
+        // Seperate by parameters
+        if query_string.contains("&") {
+            query_param = query_string.sub_string(0, query_string.index_of("&"));
+            query_string = query_string.sub_string(query_string.index_of("&") + 1);
+        } else {
+            query_param = query_string;
+            query_string.clear();
+        }
+        // set the page parameter to the value from the url
+        if query_param.sub_string(0, query_param.index_of("=")) == "page" {
+            try {
+            query_params.page = parse_int(query_param.sub_string(query_param.index_of("=") + 1));
+            } catch {
+                query_params.page = 0;
+            }
+        }
+    } while ( query_string.len() > 0)
+}
+
+// Return the query_params
+query_params
+```
+The script above currently return an object containing only the page parameter. It can be easily extended to include other parameters as required. To access the object inside handlebars template, the following can be used.
```

Comment:
```suggestion
The script above currently returns an object containing only the page parameter. It can be easily extended to include other parameters as required. To access the object inside the handlebars template, the following can be used:
```